### DISCUSSION TITLE
State of Ruby branches

### DISCUSSION THREAD

**Discussion by rgefryer** [kudrykv/latex-yearly-planner#122] [General]
Hi @kudrykv,

First of all, what a great project!  Thanks for making this planner available, I'm enjoying using it.

I'm also keen to get involved, especially with making tweaks for the Kindle Scribe.  I'm currently not very experienced with either Go or Ruby, though I'd rather work with Ruby as it aligns with some of my other goals.

So my question is whether I should be focussing on working with main, or whether either of the Ruby branches are complete/stable enough that I could start there instead?

Thanks,

Richard

**Comment by kudrykv** [comment]
<3

Unfortunately, the current ruby rewrite is of rare roasting degree. `rubify` is the first take on ruby rewrite. It is more complete, but I don't like the code that's emerging there. For this reason, I've restarted the rewrite again in `rubify2` (and borrowing some bits from `rubify`). For now, I'm following my vision there pretty closely. Hopefully, I will finish `rubify2` and it will replace `main` in the future (and `main` will become `legacy` or something like that). Currently, though, `rubify2` is work in progress as I don't have enough time prioritized for this project.

Current `main` branch is stable and tweakable, although code is shit.
